### PR TITLE
Rebased/fixed: Added cask args for support of appdir=/Applications and such.

### PR DESCRIPTION
**Pull request by Liquidsoul**
Hi!

Here is my attempt to rebase and fix #94 on `master`.
I wanted to get it out as soon as possible to have some feedback so it may still need some work :smile: 

Hope this helps!

### CODE REVIEW

**Comment by mikemcquaid** [comment]
Looks good to me if it does to @xu-cheng :+1:

**Comment by xu-cheng** [comment]
:+1: although should we squash commits?

File: lib/bundle/dsl.rb [hunk 1]
```diff
@@ -63,12 +66,21 @@ def install
...
     end
 
-    def cask(name)
-      @entries << Entry.new(:cask, name)
+    def cask(name, options={})
+      if options[:args]
+        options = @cask_args.merge(options[:args])
+      else
+        options = @cask_args
+      end
```
Comment (xu-cheng) [inline lib/bundle/dsl.rb]: You should update `options[:args]` here not `options`. In fact, you can replace if condition to:
```ruby
options[:args] = @cask_args.merge options.fetch(:args, {})

File: Readme.md [hunk 2]
```diff
@@ -40,6 +41,7 @@ Then list your Homebrew based dependencies in your `Brewfile`:
     brew 'mysql'
     cask 'google-chrome'
     cask 'java' unless system '/usr/libexec/java_home --failfast'
+    cask 'firefox', args: ['appdir=/Applications']
```
Comment (xu-cheng) [inline Readme.md]: `args: { appdir: '/Applications' }`

**Comment by Liquidsoul** [comment]
I wanted to squash the commits from the previous PR but I preferred to sent the full branch before doing this.
Do you want me to squash them and update the PR?

File: Readme.md [hunk 3]
```diff
@@ -40,6 +41,7 @@ Then list your Homebrew based dependencies in your `Brewfile`:
     brew 'mysql'
     cask 'google-chrome'
     cask 'java' unless system '/usr/libexec/java_home --failfast'
+    cask 'firefox', args: {'appdir=/Applications'}
```
Comment (xu-cheng) [inline Readme.md]: Sorry but you still fail to get it right. It should be: `{ appdir: '/Applications' }`

File: spec/caskinstaller_spec.rb [hunk 4]
```diff
@@ -57,7 +70,7 @@ def do_install
 
     context "when cask is installed" do
       before do
-        allow(Bundle::CaskInstaller).to receive(:installed_casks).and_return(["google-chrome"])
+        allow(Bundle::CaskInstaller).to receive(:installed_casks).and_return(["google-chrome","firefox"])
```
Comment (xu-cheng) [inline spec/caskinstaller_spec.rb]: Why change this?

File: spec/caskinstaller_spec.rb [hunk 5]
```diff
@@ -37,6 +41,15 @@ def do_install
...
       end
+
+      it "works fine with arguments" do
+        allow(Bundle).to receive(:cask_installed?).and_return(false, true)
+        allow(ARGV).to receive(:verbose?).and_return(false)
+        allow(Bundle::CaskInstaller).to receive(:installed_casks).and_return([])
+        expect(Bundle).to receive(:system).with("brew", "install", "caskroom/cask/brew-cask").and_return(true)
+        expect(Bundle).to receive(:system).with("brew", "cask", "install", "firefox", "--appdir=/Applications").and_return(true)
+        expect(do_install_with_args).to eq(true)
+      end
```
Comment (xu-cheng) [inline spec/caskinstaller_spec.rb]: This test may be more suitable to put after [Line 90](https://github.com/Liquidsoul/homebrew-bundle/blob/master/spec/caskinstaller_spec.rb#L90)

File: spec/caskinstaller_spec.rb (hunk 4 above)
Comment (Liquidsoul) [inline spec/caskinstaller_spec.rb]: I don't really know. This comes from #94.
It does not seem relevant to me. I will remove this.

File: Readme.md (hunk 3 above)
Comment (Liquidsoul) [inline Readme.md]: Ok I'll fix the Readme.
I am stil quite new to ruby and this code base. So I don't know if [this code](https://github.com/Liquidsoul/homebrew-bundle/blob/master/spec/caskinstaller_spec.rb#L9) tests this correctly.

**Comment by mikemcquaid** [comment]
Would be good to squash down the commits to e.g. 1 per author.

**Comment by Liquidsoul** [comment]
I've squashed the commits of the previous PR and add the fixes following your comments.

**Pull request closed without merging**